### ISSUE TITLE
Cascading Dual Cone Optimizer

### ISSUE THREAD

**Issue by stevengogogo** [liruipeng/ml2#20]
Since DCGD is a torch.optim.Optimizer. It is possible to "cascading" multiple loss terms

```
Loss1, loss2 ->  loss12
loss3, loss4 -> loss34
loss12, loss34 -> loss
```
